src/sst/elements/merlin/topology/polarfly.h

```diff
@@ -0,0 +1,243 @@
+
+// Authors: Kartik Lakhotia, Sai Prabhakar Rao Chenna
+
+
+#ifndef COMPONENTS_MERLIN_TOPOLOGY_POLARFLY_H
+#define COMPONENTS_MERLIN_TOPOLOGY_POLARFLY_H
+
+#include <sst/core/event.h>
+#include <sst/core/link.h>
+#include <sst/core/params.h>
+#include <sst/core/rng/rng.h>
+
+#include <unistd.h>
+#include <string.h>
+#include <vector>
+#include <algorithm>
+#include <stdlib.h>
+#include <iostream>
+#include <fstream>
+#include <sstream>
+
+#include "sst/elements/merlin/router.h"
+
+
+namespace SST {
+namespace Merlin {
+
+
+//For now, polarfly_event is not much different from the router event. Just added the hop_count metric to record the total number of hops this packet is taking across the network
+class topo_polarfly_event : public internal_router_event {
+public:
+
+    int hop_count; //Count no of hops this packet has taken so far
+     
+    bool non_minimal;
+    int valiant;
+
+    topo_polarfly_event() {}
+
+    topo_polarfly_event(int hcount) : non_minimal(false) {	
+        hop_count = hcount;
+    }
+
+    virtual ~topo_polarfly_event() {}
+
+    virtual internal_router_event* clone(void) override
+    {
+        return new topo_polarfly_event(*this);
+    }
+
+    void serialize_order(SST::Core::Serialization::serializer &ser)  override {
+        internal_router_event::serialize_order(ser);
+        
+        ser & hop_count;
+        ser & non_minimal;
+        ser & valiant;
+    }
+
+protected:
+
+private:
+    ImplementSerializable(SST::Merlin::topo_polarfly_event)
+
+};
+
+
+class topo_polarfly_init_event : public topo_polarfly_event {
+public:
+    //Insert code
+    int phase;
+
+    //track what is covered
+    int total_routers;
+    std::vector<int> covered;
+
+    topo_polarfly_init_event() {}
+
+    topo_polarfly_init_event(int p, int N) : topo_polarfly_event(0), phase(p), total_routers(N) { covered.resize(N, 0); }
+
+    virtual ~topo_polarfly_init_event() { }
+
+    virtual internal_router_event* clone(void) override
+    {
+        topo_polarfly_init_event* tte = new topo_polarfly_init_event(*this);
+        //Insert code
+        return tte;
+        
+    }
+
+    void serialize_order(SST::Core::Serialization::serializer &ser)  override {
+        topo_polarfly_event::serialize_order(ser);
+        ser & phase;
+        ser & total_routers;
+        for (int i=0; i<total_routers; i++)
+            ser & covered[i];
+    }
+private:
+    ImplementSerializable(SST::Merlin::topo_polarfly_init_event)
+
+};
+
+
+class topo_polarfly: public Topology {
+
+public:
+
+    SST_ELI_REGISTER_SUBCOMPONENT_DERIVED(
+        topo_polarfly,
+        "merlin",
+        "polarfly",
+        SST_ELI_ELEMENT_VERSION(1,0,0),
+        "PolarFly: A Cost-effective Two-diameter highly scalable topology object",
+        SST::Merlin::Topology)
+
+    SST_ELI_DOCUMENT_PARAMS(
+        // Parameters needed for use with old merlin python module
+        {"polarfly.q", "size of the Gallois field. parameter should be a prime power."},
```

Comment:
The polarfly.* params are not needed.  I'm sure you copied this from existing topologies, but these are in those topologies for backward compatibility to an older python module implementation.  They won't be needed for this one.  All of these should be removed.